### ISSUE TITLE
Minor consensus "bug" (incorrect comment/assumption)

### ISSUE THREAD

**Issue by MicahZoltu** [ethereum/go-ethereum#26628]
https://github.com/ethereum/go-ethereum/blob/91cb6f863a965481e51d5d9c0e5ccd54796fd967/core/types/transaction.go#L151-L160

First byte in range `[0, 0x7f]` are 2718 transactions, but not every first byte in the range `[0x80, 0xff]` is a legacy transaction.  While I suspect that `rlp.DecodeBytes` will correctly error on any invalid legacy transaction today, there is a very marginal risk that some future change could result in transactions with a leading byte outside of the `[0xc0, 0xfe]` not failing the `rlp.DecodeBytes` function at which point a consensus bug could occur.

At the least, the comment should probably be updated to make it clear that not all transactions in the `[0x80, 0xff]` range are legacy transactions.

Note: I recognize this is *incredibly* minor and arguably nitpicky, so I won't be offended if it gets closed without being fixed.

**Comment by karalabe** [comment]
Myeah, this is too nitpicky. The id range for new txs were explicitly chosen not to overlap with the RLP encoding. It is a very deliberate decision so I don't think it will just be forgotten. RLP itself is kind of "done", with no changes ever being made to make parsing more relaxed. And we want to switch everything over to SSZ soon, so RLP itself for txs will die out pretty fast.

TL;DR: There's not much point to tweak code that works and has a limited shelf life at this point :) But thanks for the notice!
